bundles/framework/maplegend/publisher/MapLegendHandler.js

```diff
@@ -0,0 +1,68 @@
+
+import { StateHandler, controllerMixin } from 'oskari-ui/util';
+
+class UIHandler extends StateHandler {
+    constructor (initialData, sandbox, tool) {
+        super();
+        this.tool = tool;
+        this.sandbox = sandbox;
+        this.setState({
+            showLegends: initialData || false
+        });
+        this.eventHandlers = this.createEventHandlers();
+    };
+
+    getName () {
+        return 'MapLegendHandler';
+    }
+
+    isDisplayed () {
+        return this.sandbox.findAllSelectedMapLayers().some(l => l.getLegendImage());
+    }
+
+    setShowLegends (value) {
+        this.updateState({
+            showLegends: value
+        });
+        if (value === false) {
+            this.tool.setEnabled(false);
+        } else {
+            this.tool.setEnabled(true);
+        }
```

Comment:
This feels wrong having the state in two places. I'm not sure if we could do something about it since it's bound to go out of sync at some point. We could atleast lose the if and just call `this.tool.setEnabled(!!value);`